src/main/kotlin/de/bund/digitalservice/useid/identification/IdentificationSessionController.kt

```diff
@@ -81,94 +92,86 @@ class IdentificationSessionsController(
+    )
+    fun getTCToken(@PathVariable useIdSessionId: UUID): ResponseEntity<TCTokenType> {
+        return try {
+            val identificationSession = identificationSessionService.findByUseIdSessionId(useIdSessionId)
+                ?: return ResponseEntity.status(HttpStatus.NOT_FOUND).build()
+            val eidService = EidService(eidServiceConfig, identificationSession.requestDataGroups)
+            val tcToken = eidService.getTcToken("${applicationProperties.baseUrl}$REFRESH_PATH")
+            val eIdSessionId = UriComponentsBuilder
+                .fromHttpUrl(tcToken.refreshAddress)
+                .encode().build()
+                .queryParams.getFirst("sessionId")
+            identificationSessionService.updateEIDSessionId(useIdSessionId, UUID.fromString(eIdSessionId))
+            tcTokenCallsSuccessfulCounter.increment()
+            ResponseEntity
+                .status(HttpStatus.OK)
+                .contentType(MediaType.APPLICATION_XML)
+                .body(tcToken)
+        } catch (e: Exception) {
+            tcTokenCallsWithErrorsCounter.increment()
+            log.error("Failed to get tc token for identification session. useIdSessionId=$useIdSessionId", e)
+            throw e
+        }
     }
 
     @GetMapping("/{eIdSessionId}", produces = [MediaType.APPLICATION_JSON_VALUE])
     @Operation(summary = "Fetch data as eService after identification was successful")
     @ApiResponse(responseCode = "200")
-    @ApiResponse(responseCode = "404", description = "No corresponding session found for that eIdSessionId", content = [Content()])
-    @ApiResponse(responseCode = "401", description = "Authentication failed (missing or wrong api key)", content = [Content()])
+    @ApiResponse(
+        responseCode = "404",
+        description = "No corresponding session found for that eIdSessionId",
+        content = [Content()]
+    )
+    @ApiResponse(
+        responseCode = "401",
+        description = "Authentication failed (missing or wrong api key)",
+        content = [Content()]
+    )
     @SecurityRequirement(name = "apiKey")
-    fun getIdentity(@PathVariable eIdSessionId: UUID, authentication: Authentication): Mono<ResponseEntity<GetResultResponseType>> {
-        /*
-            Wrapping blocking call to the SDK into Mono.fromCallable
-            https://projectreactor.io/docs/core/release/reference/index.html#faq.wrap-blocking
-        */
+    fun getIdentity(
+        @PathVariable eIdSessionId: UUID,
+        authentication: Authentication
+    ): ResponseEntity<GetResultResponseType> {
         val apiKeyDetails = authentication.details as ApiKeyDetails
-        val getIdentityResult = Mono.fromCallable {
+
+        val userData: GetResultResponseType?
+        val identificationSession: IdentificationSession?
+        identificationSession = identificationSessionService.findByEIDSessionId(eIdSessionId)
+            ?: return ResponseEntity.status(HttpStatus.NOT_FOUND).build()
```

Comment:
A bit odd that we init a variable and then assign it later in the next line
```suggestion
 val userData: GetResultResponseType
        val identificationSession: IdentificationSession = identificationSessionService.findByEIDSessionId(eIdSessionId)
            ?: return ResponseEntity.status(HttpStatus.NOT_FOUND).build()
```

Reply:
as suggested by intelliJ as well.

Reply:
Done